### PR TITLE
{Compute} Fix tests to disable default outbound access

### PR DESCRIPTION
**Pull request by yanzhudd**
**Related command**

`az vm create --subnet mySubnet --vnet-name myVnet`
`az network vnet subnet update --default-outbound-access false --subnet mySubnet --vnet-name myVnet`
or `az vm create --public-ip-sku Standard`

`az vmss create --orchestration-mode Flexible`
or `az vmss update --lb-sku Standard`
or `az vmss create --lb-sku Standard`

**Description**

Relative PR:
https://github.com/Azure/azure-cli/pull/29181
https://github.com/Azure/azure-cli/pull/29232
https://github.com/Azure/azure-cli/pull/29287
#29316

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 1]
```diff
@@ -5976,43 +6140,62 @@ def test_vmss_disk_encryption_e2e(self, resource_group, resource_group_location,
...
+
+        # Disable default outbound access
+        self.cmd(
+            'network vnet subnet update -g {rg} --vnet-name {vnet} -n {subnet} --default-outbound-access false')
+
         self.cmd('vm show -g {rg} -n {vm}', checks=[
             self.check('securityProfile.encryptionAtHost', True)
         ])
-        self.cmd('vmss create -g {rg} -n {vmss} --image OpenLogic:CentOS:7.5:latest --generate-ssh-keys --encryption-at-host --admin-username exampleusername')
+        self.cmd('vmss create -g {rg} -n {vmss} --image OpenLogic:CentOS:7.5:latest --generate-ssh-keys --encryption-at-host --admin-username exampleusername --orchestration-mode Flexible')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:6198]: >  --orchestration-mode Flexible

Why do we need to add it?

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 2]
```diff
@@ -5960,7 +6124,7 @@ def test_vmss_disk_encryption_e2e(self, resource_group, resource_group_location,
         self.kwargs.update({
             'vmss': 'vmss1'
         })
-        self.cmd('vmss create -g {rg} -n {vmss} --image Win2022Datacenter --instance-count 1 --admin-username clitester1 --admin-password Test123456789! --orchestration-mode Uniform')
+        self.cmd('vmss create -g {rg} -n {vmss} --image Win2022Datacenter --instance-count 1 --admin-username clitester1 --admin-password Test123456789! --orchestration-mode Uniform --lb-sku Standard')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:6127]: > --lb-sku Standard

Why do we need to add it?

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py (hunk 1 above)
Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:6198]: There are several ways to disable outbound access. For VMSS, one way is to use the Flexible orchestration mode.

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py (hunk 2 above)
Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:6127]: There are several ways to disable outbound access. For VMSS, one way is to use Standard load balancer.

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:6198]: Got it, thanks!

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py (hunk 2 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:6127]: Got it, thanks!

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**